Code/GraphMol/FileParsers/MultithreadedSDMolSupplier.h

```diff
@@ -30,7 +30,7 @@ class RDKIT_FILEPARSERS_EXPORT MultithreadedSDMolSupplier
       const MolFileParserParams &parseParams = MolFileParserParams());
 
   MultithreadedSDMolSupplier();
-  ~MultithreadedSDMolSupplier() override;
+  virtual ~MultithreadedSDMolSupplier() {close();}
```

Comment:
I think the dtor here only needs to be virtual if you are planning on inheriting from this class, which I don't think we are.

Reply:
I believe that at least some of this is not true.
If you derive from a class, then the dtor of the base needs to be virtual, but the derived class does not need to have a virtual dtor.
Simple example:
```
#include <iostream>

class Base {
 public:
  virtual ~Base() { std::cerr << "Base destructor called" << std::endl; };
};

class Derived : public Base {
 public:
  ~Derived() override {
    std::cerr << "Derived destructor called" << std::endl;
  };
};

int main() {
  Base* b = new Derived();
  std::cerr << "delete b;" << std::endl;
  delete b;  // This will call the Derived destructor, followed by the Base
             // destructor
}
```

This outputs:
```
delete b;
Derived destructor called
Base destructor called
```
with both g++ and MSVC++

Reply:
But this is semantics, in this case they both have the same behavior so if it bugs you I can remove it.